core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -138,26 +139,18 @@ object ByteBufferMessageSet {
       }
 
       override def makeNext(): MessageAndOffset = {
-        messageAndOffsets match {
-          // Using inner offset and timestamps
-          case Some(innerMessageAndOffsets) =>
-            innerMessageAndOffsets.pollFirst() match {
-              case null => allDone()
-              case MessageAndOffset(message, offset) =>
-                val relativeOffset = offset - lastInnerOffset
-                val absoluteOffset = wrapperMessageOffset + relativeOffset
-                new MessageAndOffset(message, absoluteOffset)
-            }
-          // Not using inner offset and timestamps
-          case None =>
-            try readMessageFromStream()
-            catch {
-              case eofe: EOFException =>
-                compressed.close()
-                allDone()
-              case ioe: IOException =>
-                throw new KafkaException(ioe)
+        val nextMessage = messageAndOffsets.pollFirst()
+        nextMessage match {
+          case null => allDone()
+          case MessageAndOffset(message, offset) => {
```

Comment:
You can do (no curly braces needed):

```scala
case nextMessage@ MessageAndOffset(message, offset) =>
```

And then you don't need to declare the `nextMessage` val outside the `match`.